snippets/object-from-key-value-pairs.md

```diff
@@ -4,5 +4,5 @@ Use `map()` to create objects for each key-value pair, combine with `Object.assi
 
 ```js
 var objectFromPairs = arr =>
-  Object.assign(...arr.map( v => {return {[v[0]] : v[1]};} ));
+  Object.assign(...arr.map( v => ({ [v[0]] : v[1] })));
```

Comment:
This is indeed shorter, but sacrifices readability for shorter code. I am not entirely sold on the idea. Any other opinions?

Reply:
How about: 

`var objectFromPairs = arr => arr.reduce((a,b) => (a[b[0]] = b[1], a), {})`

or 

`var objectFromPairs = arr => arr.reduce((a,b) => { a[b[0]] = b[1]; return a;}, {})`

Reply:
The first variant is quite easy to read. Is it any faster performance-wise than the original one? (just curious) I would definitely prefer the `.reduce()` variant, though in terms of readability as even the original one I had implemented felt wrong to me somehow.

Reply:
Although the ES6 version is an interesting implementation, I feel that the `.reduce()` version is more readable. I would assume that the `.reduce()` version would perform marginally better since it doesn't create intermediary arrays and objects.

Reply:
```

Reply:
Right, so we'll stick with `reduce()`.